cpp/core/jni/JniWrapper.cc

```diff
@@ -541,16 +540,26 @@ Java_io_glutenproject_vectorized_NativeColumnarToRowJniWrapper_nativeConvertColu
   auto backend = gluten::createBackend();
   std::shared_ptr<ColumnarToRowConverter> columnarToRowConverter =
       gluten::jniGetOrThrow(backend->getColumnar2RowConverter(allocator, cb));
-  gluten::jniAssertOkOrThrow(
-      columnarToRowConverter->init(),
-      "Native convert columnar to row: Init "
-      "ColumnarToRowConverter failed");
-  gluten::jniAssertOkOrThrow(
-      columnarToRowConverter->write(), "Native convert columnar to row: ColumnarToRowConverter write failed");
+  int64_t instanceID = columnarToRowConverterHolder.insert(columnarToRowConverter);
+  return instanceID;
+  JNI_METHOD_END(-1)
+}
+
+JNIEXPORT jobject JNICALL
+Java_io_glutenproject_vectorized_NativeColumnarToRowJniWrapper_nativeColumnarToRowWrite( // NOLINT
+    JNIEnv* env,
+    jobject,
+    jlong batchHandle,
+    jlong instanceId) {
+  JNI_METHOD_START
+  auto columnarToRowConverter = columnarToRowConverterHolder.lookup(instanceId);
+  std::shared_ptr<ColumnarBatch> cb = glutenColumnarbatchHolder.lookup(batchHandle);
+  GLUTEN_THROW_NOT_OK(columnarToRowConverter->write(cb));
 
   const auto& offsets = columnarToRowConverter->getOffsets();
   const auto& lengths = columnarToRowConverter->getLengths();
-  int64_t instanceID = columnarToRowConverterHolder.insert(columnarToRowConverter);
+
+  auto numRows = cb->getNumRows();
```

Comment:
Where to use `numRows`?

Reply:
https://github.com/JkSelf/gluten/blob/9e0edbedff2a67146b98b41e3cff971c215b247c/cpp/core/jni/JniWrapper.cc#L566  and https://github.com/JkSelf/gluten/blob/9e0edbedff2a67146b98b41e3cff971c215b247c/cpp/core/jni/JniWrapper.cc#L569 use the `numRows`.